registry/remote/registry_example_test.go

```diff
@@ -0,0 +1,79 @@
+package remote_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"testing"
+
+	"oras.land/oras-go/v2/registry/remote"
+)
+
+func testRegistry() *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		result := struct {
+			Repositories []string `json:"repositories"`
+		}{
+			Repositories: []string{"public/repo1", "public/repo2", "internal/repo3"},
+		}
+		json.NewEncoder(w).Encode(result)
+	}))
+}
+
+var exampleReg *remote.Registry
```

Comment:
It is better to have name `exampleRegistry` instead of its short name `exampleReg`.